components/tidb_query_normal_expr/src/builtin_encryption.rs

```diff
@@ -384,4 +403,25 @@ mod tests {
             Datum::Null
         );
     }
+
+    #[test]
+    fn test_password() {
+        let cases = vec![
+            ("TiKV", "*cca644408381f962dba8dfb9889db1371ee74208"),
+            ("Pingcap", "*f33bc75eac70ac317621fbbfa560d6251c43cf8a"),
+            ("rust", "*090c2b08e0c1776910e777b917c2185be6554c2e"),
+            ("database", "*02e86b4af5219d0ba6c974908aea62d42eb7da24"),
+            ("raft", "*b23a77787ed44e62ef2570f03ce8982d119fb699"),
+        ];
+        let mut ctx = EvalContext::default();
+
+        for (input_str, exp_str) in cases {
+            let input = datum_expr(Datum::Bytes(input_str.as_bytes().to_vec()));
+            let op = scalar_func_expr(ScalarFuncSig::Password, &[input]);
+            let op = Expression::build(&mut ctx, op).unwrap();
+            let got = op.eval(&mut ctx, &[]).unwrap();
+            let exp = Datum::Bytes(exp_str.as_bytes().to_vec());
+            assert_eq!(got, exp, "password('{:?}')", input_str);
```

Comment:
It could be better to add testing for empty input and the warning I think :)

Reply:
@Fullstop000 I have added a Null case test, similar to the above tests. PTAL.